File: chart_server_fe/src/main/kotlin/io/madrona/njord/components/NavBar.kt
package io.madrona.njord.components

import kotlinx.html.ButtonType
import react.Props
import react.dom.*
import react.fc
import react.router.dom.Link

val NavBar = fc<Props> {
    nav(classes = "navbar navbar-expand-lg navbar-dark bg-dark") {
        div(classes = "container-fluid") {
            Link {
                + "Njord"
                attrs.also {
                    it.className = "navbar-brand"
                    it.to = "/v1/app"
                }
            }
            button(classes = "navbar-toggler", type = ButtonType.button) {
                setProp("data-bs-toggle", "collapse")
                setProp("data-bs-target", "#navbarNavAltMarkup")
                setProp("aria-expanded", false)
                span(classes = "navbar-toggler-icon") {}
            }
            div(classes = "collapse navbar-collapse") {
                setProp("id", "navbarNavAltMarkup")
                div(classes = "navbar-nav") {
                    Link {
                        + "About"
                        attrs.also {
                            it.className = "navbar-text"
                            it.to = "/v1/app/about"
                        }
                    }
                }
            }
        }
    }
}

File: chart_server_fe/src/NavBar.js
import React from "react";
import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import {Link} from "react-router-dom";

function NavLink(props) {
    return (
        <li className="nav-item"><Link to={props.path} className="nav-link">{props.label}</Link></li>
    )
}
function NavBar() {
    return (
        <Navbar bg="secondary" expand="lg" variant="dark">
            <Container>
                <Navbar.Brand>Njord ENC Server</Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav" />
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="me-auto">
                        <NavLink path="/" label="About" />
                        <NavLink path="/enc" label="ENC (Electronic Navigation Chart)" />
                        <NavLink path="/control/charts" label="Control Panel" />
                    </Nav>
                </Navbar.Collapse>
            </Container>
        </Navbar>
    );
}

export default NavBar;
